src/sanitise-old-messages/src/sanitiseOldMessages.ts

```diff
@@ -0,0 +1,103 @@
+import { add } from "date-fns"
+import type { Client } from "pg"
+import { logger } from "shared"
+import type { ApiClient, AuditLog, AuditLogDynamoGateway, PromiseResult } from "shared-types"
+import { isError } from "shared-types"
+import type { SanitiseOldMessagesConfig } from "./config"
+
+export default class SanitiseOldMessages {
+  constructor(
+    private api: ApiClient,
+    private dynamo: AuditLogDynamoGateway,
+    private db: Client,
+    private config: SanitiseOldMessagesConfig
+  ) {
+    this.api = api
+    this.dynamo = dynamo
+    this.db = db
+    this.config = config
+  }
+
+  public async sanitiseOldMessages() {
+    logger.debug("Fetching messages to sanitise")
+
+    // Fetch messages over the configured age threshold from dynamo
+    const messages = await this.dynamo.fetchUnsanitised(this.config.MESSAGE_FETCH_BATCH_NUM)
+    if (isError(messages)) {
+      logger.error({ message: "Unable to fetch messages from dynamo, exiting", error: messages })
+      return
+    }
+
+    // Call postgres and check if we should sanitise each message
+    for (const message of messages) {
+      const shouldSanitiseResult = await this.shouldSanitise(message, this.config.SANITISE_AFTER)
+      if (isError(shouldSanitiseResult) || shouldSanitiseResult.error) {
+        logger.error({
+          message: "Unable to check if message is sanitised",
+          error: shouldSanitiseResult,
+          messageId: message.messageId
+        })
+        continue
+      } else if (shouldSanitiseResult.sanitise) {
+        const sanitiseResult = await this.api.sanitiseMessage(message.messageId)
```

Comment:
Suggestion: We could delete the nextSanitiseCheck field in dynamo, either here or in the API call, which would remove sanitised items from the sparse index and reduce the size of the index by quite a lot https://stackoverflow.com/questions/28258024/dynamodb-query-on-boolean-key

Reply:
This is a fantastic suggestion and is the culmination of a lot of pain - excellent job! 💯 

I've applied this to:
- The `audit-log-api` to remove the `nextSanitiseCheck` field when sanitising an audit log
- The `sanitise-old-messages` lambda to stop updating `nextSanitiseCheck` if we've successfully sanitised an audit log
- `DynamoGateway` and `AuditLogDynamoGateway` to remove the `updateEntryUnconditionally` method, since there's no need for it anymore

Reply:
Great idea Alice! 👌 This definitely reduces the size as we are projecting all fields in our indexes. (Every record in the main table is duplicated in each index)

Reply:
Done this. Tested on sandbox successfully 🎉